packages/storage/src/cookies.ts

```diff
@@ -52,20 +51,24 @@ function deserializeCookieOptions(cookie: string, key: string) {
   return cookie.match(`(^|;)\\s*${key}\\s*=\\s*([^;]+)`)?.pop() ?? null;
 }
 
-let useRequest: () => PageEvent | undefined;
-try {
-  useRequest = require("solid-start/server").useRequest;
-} catch (e) {
-  useRequest = () => {
-    // eslint-disable-next-line no-console
-    console.warn(
-      "It seems you attempt to use cookieStorage on the server without having solid-start installed or use vite.",
+const getRequestHeaders = isServer
+  ? () => getRequestEvent()?.request?.headers || new Headers()
+  : () => new Headers();
+const setCookie = isServer
+  ? (key: string, value: string, options: CookieProperties) => {
+    const responseHeaders = (
+      getRequestEvent() as unknown as RequestEvent & { response?: { headers: Headers } }
+    )?.response?.headers;
+    if (!responseHeaders) {
+      return;
+    }
+    const serializedOptions = serializeCookieOptions(options);
+    responseHeaders.append(
```

Comment:
I have a bad feeling there, because we make the set cookie string indefinitely longer each time a vhange happens neither in the deserialization nor in the serialization we remove the unnecessary values, we should definitely consider cleaning up the set cookies header.

Reply:
Good point. That's why I originally switched to set.